lld/COFF/DLL.cpp

```diff
@@ -142,6 +142,30 @@ class NullChunk : public NonSectionChunk {
   size_t size;
 };
 
+// A chunk for ARM64EC auxiliary IAT.
+class AuxImportChunk : public NonSectionChunk {
+public:
+  explicit AuxImportChunk(ImportFile *file) : file(file) {
+    setAlignment(sizeof(uint64_t));
+  }
+  size_t getSize() const override { return sizeof(uint64_t); }
+
+  void writeTo(uint8_t *buf) const override {
+    uint64_t impchkVA = 0;
+    if (file->impchkThunk)
+      impchkVA = file->impchkThunk->getRVA() + file->ctx.config.imageBase;
```

Comment:
Hmm, so these aux IAT entries contain a full absolute address when on disk? That's unusual, but I guess it doesn't matter (wrt requiring runtime base relocations or not) as the loader needs to fill them in, in any case.

Reply:
Yes, my guess is that it was just more convenient to use regular base relocations.